presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -273,6 +267,27 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         return concat(partitionBatches);
     }
 
+    private void ensureBucketingMatch(HiveBucketProperty tableBucketProperty, HivePartition hivePartition, Partition partition)
```

Comment:
I would pass in `(Optional<HiveBucketProperty> tableBucketProperty, Optional<HiveBucketProperty> partitionBucketProperty, HivePartition hivePartition)`, and move `if (bucketProperty.isPresent())` check into this method.

One can probably argue either way. I prefer having the logic about bucketing validation in one place.